tests/rws_tests.py

```diff
@@ -9,7 +9,26 @@
 from RwsSet import RwsSet
 from RwsCheck import RwsCheck
 from RwsCheck import WELL_KNOWN
-from check_sites import find_diff_sets
+from check_sites import find_diff_sets, load_rws_file_as_json
+
+class TestLoadFile(unittest.TestCase):
+    """A test suite for the load_rws_file_as_json function"""
+
+    def test_load_only(self):
+        self.assertEqual(load_rws_file_as_json("tests/not_json.JSON", False),
+                         (None, "There was an error when loading tests/not_json.JSON;\nerror was:  Expecting value: line 1 column 1 (char 0)"))
+        self.assertEqual(load_rws_file_as_json("tests/bad_formatting.JSON", False),
+                         ({"a": "foo", "b": "bar"}, None))
+        self.assertEqual(load_rws_file_as_json("tests/good_formatting.JSON", False),
+                         ({"a": "foo", "b": "bar"}, None))
+
+    def test_load_and_format(self):
+        self.assertEqual(load_rws_file_as_json("tests/not_json.JSON", True),
+                         (None, "There was an error when loading tests/not_json.JSON;\nerror was:  Expecting value: line 1 column 1 (char 0)"))
+        self.assertEqual(load_rws_file_as_json("tests/bad_formatting.JSON", True),
+                         (None, 'Formatting for tests/bad_formatting.JSON is incorrect;\nerror was:\n-   "a": "foo", \n?              -\n+   "a": "foo",\n-     "b": "bar"\n? --\n+   "b": "bar"\n-   '))
```

Comment:
IMO it'd be nice if we could test this without having to write test-only files for it, i.e. if we could just pass in the string contents directly. That makes the tests faster (since they don't have to spin a disk) and easier to write (since they don't require new files in the repo).

The downside is that you have to split the file-reading part out of the function that verifies the content, but IMO that's not actually a bad thing. E.g.:

```python
    try:
        raw_content = Path(input_file).read_text()
    except Exception as inst:
        # If the file cannot be loaded, we will not run any other checks
        print(f"There was an error when loading {input_file};" 
              f"\nerror was:  {inst}")
        return

    (rws_sites, error) = parse_content_as_json(raw_content, with_format)
    if rws_sites == None or error != None:
        print(error)
        return
```